src/firefox/index.js

```diff
@@ -17,6 +17,82 @@ import {default as defaultFirefoxConnector, REMOTE_PORT} from './remote';
 
 const log = createLogger(__filename);
 
+// Flow Types
+
+import type {DefaultFirefoxConnectorFn} from './remote';
+import type {ChildProcess} from 'child_process';
+import type {
+  PreferencesAppName,
+  PreferencesGetterFn,
+} from './preferences';
+
+import type {ExtensionManifest} from '../util/manifest';
+
+export type FirefoxRunnerParams = {
+  binary?: string,
+  profile?: string,
+  "new-instance"?: boolean,
+  "no-remote"?: boolean,
+  "foreground"?: boolean,
+  "listen"?: number,
+  "binary-args"?: Array<string> | string,
+  "env"?: {
+    [key: string]: string
+  },
+  "verbose"?: boolean,
+};
+
+export type FirefoxRunnerResults = {
+  process: ChildProcess,
+  binary: string,
+  args: Array<string>,
+};
+
+export type FirefoxRunnerFn =
+  (params: FirefoxRunnerParams) => FirefoxRunnerResults;
```

Comment:
I guess Flow never showed an error for your declaration because returning a value (not a promise) midway through the promise chain will just pass along that value transparently.